spark/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -119,9 +120,18 @@
     }
 
     // look for split behavior overrides in options
-    this.splitSize = options.get("split-size").map(Long::parseLong).orElse(null);
-    this.splitLookback = options.get("lookback").map(Integer::parseInt).orElse(null);
-    this.splitOpenFileCost = options.get("file-open-cost").map(Long::parseLong).orElse(null);
+    this.splitSize = options.get("split-size").map(Long::parseLong).orElse(
+        Optional.ofNullable(table.properties().get(TableProperties.SPLIT_SIZE))
```

Comment:
Can you use `PropertyUtil.propertyAsLong` instead of more `Optional`? That's a bit more concise.

Reply:
Sure, will change.